SafeNeighbourhood/Pages/Index.cshtml.cs

```diff
@@ -15,13 +17,38 @@ public void OnGet()
         {
             using (var webClient = new WebClient())
             {
-                string propertyData = webClient.DownloadString("https://data.cincinnati-oh.gov/resource/w3jp-dfxy.json");
+                string propertyData = GetData("https://data.cincinnati-oh.gov/resource/w3jp-dfxy.json"); // using a method here for flexibility
 
-                var property = VacantProperty.FromJson(propertyData);
-                ViewData["property"] = property;
+                JSchema schema = JSchema.Parse(System.IO.File.ReadAllText("Schema/VacantProperty.json"));
+                JArray JsonArray = JArray.Parse(propertyData);
+
+                IList<string> validationEvents = new List<string>();
+
+                if (JsonArray.IsValid(schema))
+                {
+                    VacantProperty[] property = VacantProperty.FromJson(propertyData);
+                    ViewData["property"] = property;
+                } else
+                {
+                    foreach (string evt in validationEvents)
+                    {
+                        Console.WriteLine(evt);
+                    }
+                    ViewData["property"] = new List<VacantProperty>();
+                }
 
             }
 
         }
+
+        public string GetData(string endpoint)
+        {
+            string downloadedData = "";
+            using (WebClient webClient = new WebClient())
+            {
+                downloadedData = webClient.DownloadString(endpoint);
+            }
+            return downloadedData;
+        }
```

Comment:
The project for now is consuming only one JSON feed, but they plan on consuming two more JSON feeds. Thus, it is better to create a method for same. It reduces technical debt.